src/engine/ruleset/decoders/sonicwall/firewall-general.yml

```
@@ -0,0 +1,63 @@
+---
+name: decoder/firewall-general/0
+
+metadata:
+  module: firewall
+  title: Firewall logs decoder
+  description: Decoder for Linux firewall logs
+  author:
+    name: Wazuh Inc. info@wazuh.com
+    date: 2023-01-12
+  references:
+    - https://www.sonicwall.com/es-mx/support/technical-documentation
+
+sources:
+  - decoder/queue-localfile/0
+
+parse:
+  logpar:
+    # Jan  3 13:45:36 192.168.5.1 id=firewall sn=000SERIAL time="2007-01-03 14:48:06" fw=89.160.20.156 pri=6 c=262144 m=98 msg="Connection Opened" n=23419 src=2.2.2.2:36701:WAN dst=89.160.20.156:50000:WAN proto=tcp/50000
+    - event.original: >-
+        <~log.timestamp/date/%b %d %T> <host.ip> id=<~log.id> sn=<~log.sn> time="<@timestamp/%F %H:%M:%6S>" fw=<~log.fw>
+        pri=<~log.pri> c=<~log.c> m=<event.code> msg="<event.action>" n=<~log.n> src=<source.ip>:<source.port>:<network.type>
+        dst=<~log.dst> proto=<network.protocol>/<~log.proto>
+    - event.original: >-
+        <~log.timestamp/date/%b %d %T> <host.ip> id=<~log.id> sn=<~log.sn> time="<@timestamp/%F %H:%M:%6S>" fw=<~log.fw>
+        pri=<~log.pri> c=<~log.c> m=<event.code> msg="<event.action>" n=<~log.n> src=<source.ip>:<source.port>:<network.type>
+        dst=<~log.dst>
+
+normalize:
+  - map:
+      - event.kind: event
+      - event.module: sonicwall
+      - event.dataset: sonicwall.firewall
+      - fileset.name: firewall
+      - input.type: log
+      - observer.product: Firewalls
+      - observer.type: Firewall
+      - observer.vendor: Sonicwall
+      - rsa.internal.messageid: $event.code
+      - rsa.time.event_time": $@timestamp
+      - tags: +a_append/forwarded/sonicwall.firewall
+  - check:
+      - ~log.proto: +ef_exists
+      - ~log.proto: +s_eq/50000
+    map:
+      - rsa.internal.event_desc: $event.action
+      - rsa.internal.msg: $event.action
+      - ~aux_slash: '/'
+      - rsa.network.sinterface: +s_concat/$network.type/ dst=/$~log.dst/ proto=/$network.protocol/$~aux_slash/$~log.proto
+      - observer.ingress.interface.name: $rsa.network.sinterface
+      - related.ip: +a_append/$source.ip
+      - event.action: +ef_delete
+  - check:
+      - event.action: +ef_exists
+    map:
+      - rsa.misc.action: +a_append/$event.action
+      - source.ip: +ef_delete
+      - source.port: +ef_delete
+  - map:
+      - host: +ef_delete
+      - network: +ef_delete
+      - ~log: +ef_delete
+      - ~aux_slash: +ef_delete
```

It's possible move `aux_slash` into `~log`, to delete all custom variables together.

```suggestion
      - ~log.aux_slash: '/'
      - rsa.network.sinterface: +s_concat/$network.type/ dst=/$~log.dst/ proto=/$network.protocol/$~log.aux_slash/$~log.proto
      .
      .
      - host: +ef_delete
      - network: +ef_delete
      - ~log: +ef_delete
```